docs/go-queries.md

```diff
@@ -55,7 +55,17 @@ You can now query `current_state` by using the CLI:
 
 `tctl --namespace samples-namespace workflow query -w my_workflow_id -r my_run_id -qt current_state`
 
-You can also issue a query from code using the `QueryWorkflow()` API on a Temporal client object.
+You can also issue a query from code using the `QueryWorkflow()` API on a Temporal client object as shown below.
+
+```golang
+response, err := temporal.QueryWorkflow(context.Background(), workflowID, runID, "current_state")
```

Comment:
if "current_state" refers to the workflow state, maybe we should list all the possible options that uses can enter here? wdyt?

Reply:
@tsurdilo @flossypurse I added some more info on calling `QueryWorkflow()` with extra positional args, and a section on `QueryWorkflowWithOptions()`.

Also, it doesn't look like the "Consistent Query" section's example is valid. There's no `QueryConsistencyLevel` option on `QueryWorkflowWithOptionsRequest`, here's the [relevant code in the Go SDK](https://github.com/temporalio/sdk-go/blob/6580cbe0aa41a8b515791f95c2c15bb37db1dab1/internal/internal_workflow_client.go#L888). Go prints a "unknown field 'QueryConsistencyLevel' in struct literal of type internal.QueryWorkflowWithOptionsRequest" compiler error when you use that option.